src/Runner/Runner.php

```diff
@@ -109,20 +115,92 @@ public function __construct(
         $this->cacheManager = $cacheManager;
         $this->directory = $directory ?: new Directory('');
         $this->stopOnViolation = $stopOnViolation;
+        $this->parallel = $parallel;
     }
 
     public function fix(): array
     {
-        $changed = [];
+        $finderIterator = $this->finder instanceof \IteratorAggregate
+            ? $this->finder->getIterator()
+            : $this->finder;
 
-        $finder = $this->finder;
-        $finderIterator = $finder instanceof \IteratorAggregate ? $finder->getIterator() : $finder;
         $fileFilteredFileIterator = new FileFilterIterator(
             $finderIterator,
             $this->eventDispatcher,
             $this->cacheManager
         );
 
+        if ($this->parallel > 1) {
+            return $this->fixParallel($fileFilteredFileIterator);
+        }
+
+        return $this->fixProcedural($fileFilteredFileIterator);
+    }
+
+    private function fixParallel(FileFilterIterator $fileFilteredFileIterator): array
+    {
+        if (false === \function_exists('pcntl_fork')) {
+            return $this->fixProcedural($fileFilteredFileIterator);
+        }
+
+        $changed = [];
+        $childProcesses = [];
+
+        $files = iterator_to_array($fileFilteredFileIterator);
+        $numFiles = \count($files);
+        $numPerBatch = (int) ceil($numFiles / $this->parallel);
+
+        foreach (array_chunk($files, $numPerBatch) as $batch) {
```

Comment:
can we have the process pool, instead?
imagine that we are having 100 files and 2 processes.
now, we divided files equally - 50 files to process A and 50 to process B.

now, imagine that first 50 files are 1MB each, while another 50 files are 10bytes each. with that, we would not have much gain, as whole actual work would be done anyway by process A

if we have the pool, we can feed the process with next file as soon as it finished consuming previous file. so we don't feed each process with same amount of files (not knowing how big they are and how much CPU we need to spend for each), but we ensure that each CPU is constantly working on any file that is still to be fixed.